api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -0,0 +1,102 @@
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.patientqueueing.api;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openmrs.BaseOpenmrsObjectTest;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.patientqueueing.PatientQueueingConfig;
+import org.openmrs.module.patientqueueing.api.dao.PatientQueueingDao;
+import org.openmrs.module.patientqueueing.api.impl.PatientQueueingServiceImpl;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+
+import javax.validation.constraints.AssertTrue;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a unit test, which verifies logic in UgandaEMRSyncService. It doesn't extend
+ * BaseModuleContextSensitiveTest, thus it is run without the in-memory DB and Spring context.
+ */
+public class PatientQueueingServiceTest extends BaseModuleContextSensitiveTest {
+
+    protected static final String QUEUE_STANDARD_DATASET_XML = "org/openmrs/module/patientqueueing/standardTestDataset.xml";
+
+
+    @Before
+    public void initialize() throws Exception {
+        executeDataSet(QUEUE_STANDARD_DATASET_XML);
+    }
+
+    @InjectMocks
+    PatientQueueingServiceImpl patientQueueingService;
+
+    @Mock
+    PatientQueueingDao dao;
+
+    @Before
+    public void setupMocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void savePatientQueue_shouldReturnPatientQueueWithCompletedStatus() throws IOException {
+
+        PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+
+        Patient patient = Context.getPatientService().getPatient(10000);
+        Location location = Context.getLocationService().getLocation(1);
+
+        List<PatientQueue> patientQueueList = Context.getService(PatientQueueingService.class).getPatientInQueueList(null, null, null, location, patient, PatientQueueingConfig.status.PENDING.toString());
+
+        PatientQueue patientQueue = patientQueueList.get(0);
+
+        Assert.assertEquals(patientQueue.getStatus(), PatientQueueingConfig.status.PENDING.toString());
+
+        patientQueueingService.completePatientQueue(patientQueue, PatientQueueingConfig.status.COMPLETED.toString());
+
+        PatientQueue completedPatientQueue = patientQueueingService.getPatientQueueById(patientQueue.getPatientQueueId());
+        Assert.assertNotNull(completedPatientQueue);
+        Assert.assertEquals(completedPatientQueue.getStatus(), PatientQueueingConfig.status.COMPLETED.toString());
+    }
+
+    @Test
+    public void savePatientQueue_shouldReturnPatientQueue() throws IOException {
```

Comment:
When you say it should return patient queue, are you testing savePatientQueue or getPatientQueue?

Reply:
That is testing saving patient Queue.

Reply:
Have you looked at the naming convention for tests in openmrs-core?